### ISSUE TITLE
Allow roles to be changed

### ISSUE THREAD

**Issue by StelFoog** [insektionen/cleaning-list-fe#14]
Mods and admins should be able to change roles as allowed by the backend. I.e:

- Can't change own role
- `mod` users can  change `base` and `manager` users' roles between `base` and `manager`
- `admin` users can change any users' role into any role

**Comment by Cactooz** [comment]
This is really needed! It is annoying to have create new accounts if an account has the wrong role.

**Comment by StelFoog** [comment]
Yeah it should actually have been part of the initial version, but I forgot about it since the `DetailRow` and `Select` components don't intersect well naturally 😅
